BatteryExamine.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+
+public class BatteryExamine
+{
+
+    string message = string.Empty;
+    Dictionary<string, string> batteryAlerts = new Dictionary<string, string>();
+
+    public Dictionary<string, string> BatteryIsOk(BatteryFactors batteryFactors)
+    {
+        batteryAlerts.Clear();
+        if (batteryFactors.temperatureMesureUnit == "Fahrenheit")
+        {
+            batteryFactors.batteryTemprature = BatteryFeatureConversion.ConvertTemperatureFahrenheitToCelsius(batteryFactors.batteryTemprature);
+        }
+        CompareTemperatureWithRange(batteryFactors.batteryTemprature, batteryFactors.operatingLanguage);
+        CompareStateOfChargeWithRange(batteryFactors.batteryStateOfCharge, batteryFactors.operatingLanguage);
+        CompareChargeRateWithRange(batteryFactors.batteryChargeRate, batteryFactors.operatingLanguage);
+        return batteryAlerts;
+    }
+    private void CompareTemperatureWithRange(float batteryTemperature, string operatingLanguage)
```

Comment:
How do we unit test Check********** functions ....neither it will disclose its dependencies nor returns any thing...